synapse/_scripts/register_new_matrix_user.py

```diff
@@ -44,10 +44,18 @@ def request_registration(
     shared_secret: str,
```

Comment:
> Fix it? #11426

@dklimpel I think it would! (didn't know about the issue)

I think the current solution muddies the water a bit with `inhibit_user_in_use_error` which is a configuration setting normally only for the registration availability endpoint. And using the same name for registration makes the landscape a bit more confusing and complex imo which is why I put a disclaimer that this is just meant as a one-off.

Changing the name of the option is probably all that is needed. Maybe something like `allow_conflicting_mxid`